src/connectors/hits-per-page/connectHitsPerPage.ts

```diff
@@ -4,101 +4,120 @@ import {
- *       {value: 12, label: '12 per page'},
- *       {value: 24, label: '24 per page'},
- *     ],
- *   })
- * ]);
- */
-export default function connectHitsPerPage(renderFn, unmountFn = noop) {
+export type HitsPerPageRendererOptionsItem = {
+  /**
+   * Label to display in the option.
+   */
+  label: string;
+
+  /**
+   * Number of hits to display per page.
+   */
+  value: number;
+
+  /**
+   * Indicates if it's the current refined value.
+   */
+  isRefined: boolean;
+};
+
+export type HitsPerPageConnectorParamsItem = {
+  /**
+   * Label to display in the option.
+   */
+  label: string;
+
+  /**
+   * Number of hits to display per page.
+   */
+  value: number;
+
+  /**
+   * The default hits per page on first search.
+   *
+   * @default false
+   */
+  default?: boolean;
+};
+
+export type HitsPerPageConnectorParams = {
+  /**
+   * Array of objects defining the different values and labels.
+   */
+  items: HitsPerPageConnectorParamsItem[];
+
+  /**
+   * Function to transform the items passed to the templates.
+   */
+  transformItems?: (
+    objects: HitsPerPageConnectorParamsItem[]
+  ) => HitsPerPageConnectorParamsItem[];
+};
+
+export type HitsPerPageRendererOptions<
+  THitsPerPageWidgetParams extends HitsPerPageConnectorParams = HitsPerPageConnectorParams
+> = {
+  /**
+   * Array of objects defining the different values and labels.
+   */
+  items: HitsPerPageRendererOptionsItem[];
+
+  /**
+   * Creates the URL for a single item name in the list.
+   */
+  createURL: (value: HitsPerPageRendererOptionsItem['value']) => string;
+
+  /**
+   * Sets the number of hits per page and trigger a search.
+   */
+  refine: (number: number) => any;
+
+  /**
+   * `true` if the last search contains no result.
+   */
+  hasNoResults: boolean;
+} & RendererOptions<THitsPerPageWidgetParams>;
+
+export type HitsPerPageRenderer<
+  THitsPerPageWidgetParams extends HitsPerPageConnectorParams = HitsPerPageConnectorParams
+> = Renderer<HitsPerPageRendererOptions<THitsPerPageWidgetParams>>;
+
+export type HitsPerPageWidgetFactory<
+  THitsPerPageWidgetParams extends HitsPerPageConnectorParams = HitsPerPageConnectorParams
+> = WidgetFactory<HitsPerPageConnectorParams & THitsPerPageWidgetParams>;
+
+type HitsPerPageConnector<
+  THitsPerPageWidgetParams extends HitsPerPageConnectorParams = HitsPerPageConnectorParams
+> = (
+  /**
+   * Rendering function for the custom **HitsPerPage** widget.
+   */
+  render: HitsPerPageRenderer<THitsPerPageWidgetParams>,
+
+  /**
+   * Unmount function called when the widget is disposed.
+   */
+  unmount?: () => void
+) => HitsPerPageWidgetFactory<THitsPerPageWidgetParams>;
+
+const connector: HitsPerPageConnector = function connectHitsPerPage(
+  renderFn,
+  unmountFn = noop
+) {
   checkRendering(renderFn, withUsage());
 
-  return (widgetParams = {}) => {
-    const { items: userItems, transformItems = items => items } = widgetParams;
+  return widgetParams => {
```

Comment:
Since we forward the `widgetParams` to the render function, defaulting the widget params to `{}` is different than destructuring the widget params with a `{}` fallback.

We should probably add a test to make this behavior consistent across connectors.